### DISCUSSION TITLE
tutorials/caching-subgraph-dgs/03-caching-with-caffeine

### DISCUSSION THREAD

**Discussion by giscus** [apollographql-education/content-feedback#161] [course feedback]
# tutorials/caching-subgraph-dgs/03-caching-with-caffeine

Implementing operation caching and introducing the caching library Caffeine

https://www.apollographql.com/tutorials/caching-subgraph-dgs/03-caching-with-caffeine

**Comment by Shardul555** [comment]
How are making sure that getDocumentAsync() of our class is getting executed, in my case whenever I am executing any query, it is not going inside this block, is there any change needed from graphql configuration so that it will execute CachingPreparsedDocumentProvider class whenever any query is running.
I have kept this part of code exactly same so there should not any issues wrt initialisation of beans etc,
